### ISSUE TITLE
Reporting does not work when specifying deploymentKey programmatically

### ISSUE THREAD

**Issue by Justin-Credible** [Microsoft/cordova-plugin-code-push#117]
Steps to reproduce:
1. Ensure deployment key is NOT present in `config.xml`
2. Add a new deployment via the CLI
3. In the app, call `codePush.sync(...)` with a `SyncOptions` object populated with the `deploymentKey`
4. After update check install metrics from the command line using: `code-push history MyApp MyDeployment`

Expected results: Install metrics should have updated.

Actual results: Install metrics have NOT updated.

Prior to #108, specifying the deployment key programmatically would cause a crash on iOS. #110 fixed this crash, but the way the fix was implemented will prevent install metrics from being reported. See [here](https://github.com/Microsoft/cordova-plugin-code-push/pull/110/files#diff-7cf184072214004a5cb773da18b0173dR49) and [here](https://github.com/Microsoft/cordova-plugin-code-push/pull/110/files#diff-21c28bc9b0a8cf9a55ca53cb6f66c32bR11) where the reporting code is short circuited with an early return if the deployment key is not in `config.xml`.

**Comment by geof90** [comment]
@Justin-Credible Thanks for reporting. From what I understand looking at the code, after an app is installed, it should pull the deploymentKey from the downloaded package metadata instead of the config.xml https://github.com/Microsoft/cordova-plugin-code-push/blob/master/src/android/CodePush.java#L122 so the deployment key should not be `null`. Therefore the https://github.com/Microsoft/cordova-plugin-code-push/pull/110 fix should not impact this. However, I will test this some more later in the day and see if I can repro this.

**Comment by geof90** [comment]
@Justin-Credible  Just an update that I was not able to repro this issue, and I did see install metrics being reported correctly in my CLI, with an update that I got by specifying the deployment key in the call to `sync()` but NOT the `config.xml`.

Just for more information for us, is your app an Android or iOS app? 
What version of the plugin are you using?
When does your call to `sync()` happen in your app code? Is it the `deviceReady` handler?
